docs/bugs.json

```
@@ -1,4 +1,15 @@
 [
+    {
+        "uid": "SOL-2022-3",
+        "name": "DataLocationChange",
+        "summary": "It was possible to override internal and public function while changing the data location of the parameters or return variables from ``calldata`` to ``memory`` and vice-versa. This caused invalid code to be generated when calling such a function internally through virtual function calls.",
+        "description": "When calling external functions, it is irrelevant if the data location of the parameters is ``calldata`` or ``memory``, the encoding of the data does not change. Because of that, changing the data location when overriding external functions is allowed. The compiler incorrectly also allowed a change in the data location for overriding public functions. Since public functions can be called internally as well as externally, this causes invalid code to be generated when such an incorrectly overridden function is called internally through the base contract. The caller provides a memory pointer, but the called function expects a calldata pointer or vice-versa.",
```

I'd state directly that it's being interpreted as calldata pointer and used to address calldata.
```suggestion
        "description": "When calling external functions, it is irrelevant if the data location of the parameters is ``calldata`` or ``memory``, the encoding of the data does not change. Because of that, changing the data location when overriding external functions is allowed. The compiler incorrectly also allowed a change in the data location for overriding public functions. Since public functions can be called internally as well as externally, this causes invalid code to be generated when such an incorrectly overridden function is called internally through the base contract. The caller provides a memory pointer, but the called function interprets it as a calldata pointer or vice-versa.",
```